File: Data Structures & Algorithms/subsets/submission-9.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<vector<int>> subsets(vector<int>& nums) {
        
        vector<int> curr;
        backtrack(curr, 0, nums);
        return res;
    }
    void backtrack(vector<int>&curr, int i, vector<int>&nums){
        res.push_back(curr);
        if(i == nums.size()) return;
        for(int j = i; j < nums.size(); j++){
            curr.push_back(nums[j]);
            backtrack(curr, j+1, nums);
            curr.pop_back();
        }
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-9.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        curr, res = [], []
        nums.sort()
        def dfs(i):
            if i == len(nums):
                res.append(curr.copy())
                return
            curr.append(nums[i])
            dfs(i+1)
            while i < len(nums)-1 and nums[i]  == nums[i+1]:
                i+=1
            curr.pop()
            dfs(i+1)
        dfs(0)
        return res